R/class-Link.R

```diff
@@ -37,14 +38,49 @@ Link <- setClass("Link",
     # label = "character" # coming soon
   ),
   prototype = prototype(
-    source = new("Node"),
-    target = new("Node"),
+    source = Node(),
+    target = Node(),
     weight = 1,
+    color = NULL,
     isDirected = FALSE
   ),
   validity = check_link
 )
 
+#' Link constructor
+#' 
+#' @param source The source node
+#' @param target The target node
+#' @param weight The weight of the link
+#' @param color The color of the link
+#' @param isDirected Whether the link is directed
+#' @export
+setGeneric("Link", function(source, target, weight = 1, color = NULL, isDirected = FALSE) standardGeneric("Link"), signature = c("source", "target"))
+
+#' @export
+setMethod("Link", c("Node", "Node"), function(source, target, weight = 1, color = NULL, isDirected = FALSE) {
+  new("Link", source = source, target = target, weight = weight, color = color, isDirected = isDirected)
+})
+
+#' @export
+setMethod("Link", c("character", "character"), function(source, target, weight = 1, color = NULL, isDirected = FALSE) {
+  Link(source = Node(source), target = Node(target), weight = weight, color = color, isDirected = isDirected)
+})
+
+#' @export
+setMethod("Link", c("numeric", "numeric"), function(source, target, weight = 1, color = NULL, isDirected = FALSE) {
+  Link(source = Node(source), target = Node(target), weight = weight, color = color, isDirected = isDirected)
+})
+
+#' @export
+setMethod("Link", c("NodeId", "NodeId"), function(source, target, weight = 1, color = NULL, isDirected = FALSE) {
+  Link(source = Node(source), target = Node(target), weight = weight, color = color, isDirected = isDirected)
+})
+
+#' @export
+setMethod("Link", c("missing", "missing"), function(source, target, weight = 1, color = NULL, isDirected = FALSE) {
```

Comment:
its just so that `Link()` if called for some reason, wont err but just produce an empty Link. we got this for free when we do things like `Link <- setClass("Link")` and i just dont want to lose it despite my custom constructors.